### PR TITLE
add bloom benchmarking, perf improvement from Fnv ~= 8X

### PR DESCRIPTION
**Pull request by rob-solana**
#### Problem
 Bloom using SHA256 is slow, approximately 35X slower than hashbrown HashSet, which makes using Bloom for signature caching probably not worth the RAM savings (~16X)

 #### Summary of Changes
 move Bloom to Fnv Hasher, write benchmark that looks like the Bank's Signature cache for duplicate detection

Fixes #

### CODE REVIEW

File: benches/bloom.rs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+use solana::bloom::{Bloom, BloomHashIndex};
+use solana_sdk::hash::{hash, Hash};
+use solana_sdk::signature::Signature;
+//use std::collections::HashSet;
+use hashbrown::HashSet;
+use std::hash::Hasher;
+use test::Bencher;
+
+#[bench]
+#[ignore]
```
Comment (aeyakovenko) [inline benches/bloom.rs]: why are these bench ignore?

Comment (rob-solana) [reply benches/bloom.rs]: 'cuz our infra doesn't know about 'em and crashes when presented with results for comparison?  I dunno...  maybe @mvines can answer...  this bench is only interesting in 2 dimensions and only interesting when compared to a separate baseline (hashbrown), so is ok to stay human?

**Comment by rob-solana** [comment]
my reading of the code is that no operations are performed during FnvHasher::with_key(x), so no meaningful savings from caching initialization.  I can rip the code out of FnvHasher to save some function calls, though...  think it's worth it?

**Comment by aeyakovenko** [comment]
Not that part.  ‘pos’ is called for every K. So the entire slice is rehashed every time.

**Comment by rob-solana** [comment]
ok, weird.  if I pre-hash the slice and hash for each k, I get collisions out the waz, still staring at why `hash(k)->hash(slice)` would differ from `hash(slice)->hash(k)`

**Comment by aeyakovenko** [comment]
is it because K is sequential for you?  What about a random K?

**Comment by rob-solana** [comment]
all the k are generated from FnvHashes of lastid, which is an SHA256 of all zeroes...

**Comment by rob-solana** [comment]
move convo to #2485 ?

**Pull request merged**